### PR TITLE
[DOC-507] [2024.2] [GA] Per process metrics in YBA

### PR DESCRIPTION
**Pull request by aishwarya24**
@netlify /v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics/#per-process

### CODE REVIEW

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 1]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
 | YCQL Op Latency (Avg) | The average time of DELETE, INSERT, SELECT, and UPDATE transactions, as well as other statements through the YCQL API. | An alert should be issued when latency is close to or higher than your application SLA. | ![YCQL Op Latency](/images/yp/metrics4.png) |
 | YCQL Op Latency (P99) | The average time of the top 99% of DELETE, INSERT, SELECT, and UPDATE transactions, as well as other statements through the YCQL API. | If this value is significantly higher than expected, then it might be a cause for concern and you might want to issue an alert.<br>You should check whether or not there are consistent spikes in latency. | ![YCQL Op Latency P99](/images/yp/metrics5.png) |
 
+### Per process
+
+Per process metrics provides details about the distribution of resources among YBA agents, and is currently available only from cloud providers (AWS, GCP, and Azure). The metrics presented are not intended for alert purposes, but they can be used when examining alerts on other metrics.
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:101]:
```suggestion
Per process metrics provide details about the distribution of resources among YBA agents, and is currently available only from cloud providers (AWS, GCP, and Azure). The metrics presented are not intended for alert purposes, but they can be used when examining alerts on other metrics.
```

Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:101]: currently available only for universes deployed on cloud providers?

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 2]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
+
+| Graph | Description | Example |
+| :------------------------------------ | ------------------------------------------------------------ | -------------------------------- |
+| User CPU Seconds | The percentage of CPU utilization spent by the processor dedicated to executing program instructions in user space. | ![User CPU Seconds](/images/yp/metrics106.png) |
+| System CPU Seconds | The percentage of CPU utilization spent by processor to execute system calls in kernel space on behalf of a program.| ![System CPU Seconds](/images/yp/metrics107.png) |
+| Virtual Memory | Virtual memory size for a process. | ![Virtual Memory](/images/yp/metrics108.png) |
+| RSS Memory | Resident set size (RSS) for a process (The portion of memory occupied by a process that is held in main memory (RAM)). | ![RSS Memory](/images/yp/metrics109.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:108]:
```suggestion
| RSS Memory | The resident set size (RSS) for a process. This is the portion of memory occupied by a process that is held in main memory (RAM). | ![RSS Memory](/images/yp/metrics109.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 3]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
 | YCQL Op Latency (P99) | The average time of the top 99% of DELETE, INSERT, SELECT, and UPDATE transactions, as well as other statements through the YCQL API. | If this value is significantly higher than expected, then it might be a cause for concern and you might want to issue an alert.<br>You should check whether or not there are consistent spikes in latency. | ![YCQL Op Latency P99](/images/yp/metrics5.png) |
 
+### Per process
+
+Per process metrics provides details about the distribution of resources among YBA agents, and is currently available only from cloud providers (AWS, GCP, and Azure). The metrics presented are not intended for alert purposes, but they can be used when examining alerts on other metrics.
+
+| Graph | Description | Example |
+| :------------------------------------ | ------------------------------------------------------------ | -------------------------------- |
+| User CPU Seconds | The percentage of CPU utilization spent by the processor dedicated to executing program instructions in user space. | ![User CPU Seconds](/images/yp/metrics106.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:105]:
```suggestion
| User CPU Seconds | The percentage of CPU utilization spent by the processor dedicated to executing process instructions in user space. | ![User CPU Seconds](/images/yp/metrics106.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 4]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
+| System CPU Seconds | The percentage of CPU utilization spent by processor to execute system calls in kernel space on behalf of a program.| ![System CPU Seconds](/images/yp/metrics107.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:106]:
```suggestion
| System CPU Seconds | The percentage of CPU utilization spent by processor to execute system calls in kernel space on behalf of a process. | ![System CPU Seconds](/images/yp/metrics107.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 5]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
+| Virtual Memory | Virtual memory size for a process. | ![Virtual Memory](/images/yp/metrics108.png) |
+| RSS Memory | Resident set size (RSS) for a process (The portion of memory occupied by a process that is held in main memory (RAM)). | ![RSS Memory](/images/yp/metrics109.png) |
+| PSS Memory | The Proportional Set Size (PSS) for a process (The total size of pages it has in main memory (RAM), where each page is divided by the number of processes sharing it). This metric helps to calculate the total memory occupied by multiple processes more accurately. | ![PSS Memory](/images/yp/metrics110.png) |
+| IO Read | This represents the I/O read bytes for a process, indicating the actual number of bytes retrieved from the storage layer by the process. | ![IO Read](/images/yp/metrics111.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:110]:
```suggestion
| IO Read | The I/O read bytes for a process, indicating the actual number of bytes retrieved from the storage layer by the process. | ![IO Read](/images/yp/metrics111.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 6]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
+| IO Write | This represents the I/O write bytes for a process, specifying the actual number of bytes written to the storage layer by the process. | ![IO Write](/images/yp/metrics112.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:111]:
```suggestion
| IO Write | The I/O write bytes for a process, specifying the actual number of bytes written to the storage layer by the process. | ![IO Write](/images/yp/metrics112.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 7]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
+| Open files | Number of open files for process. | ![IO Write](/images/yp/metrics113.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:112]:
```suggestion
| Open files | Number of open files for the process. | ![IO Write](/images/yp/metrics113.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md [hunk 8]
```diff
@@ -96,6 +96,21 @@ The following tables describe metrics available via the YugabyteDB Anywhere UI.
...
+
+| Graph | Description | Example |
+| :------------------------------------ | ------------------------------------------------------------ | -------------------------------- |
+| User CPU Seconds | The percentage of CPU utilization spent by the processor dedicated to executing program instructions in user space. | ![User CPU Seconds](/images/yp/metrics106.png) |
+| System CPU Seconds | The percentage of CPU utilization spent by processor to execute system calls in kernel space on behalf of a program.| ![System CPU Seconds](/images/yp/metrics107.png) |
+| Virtual Memory | Virtual memory size for a process. | ![Virtual Memory](/images/yp/metrics108.png) |
+| RSS Memory | Resident set size (RSS) for a process (The portion of memory occupied by a process that is held in main memory (RAM)). | ![RSS Memory](/images/yp/metrics109.png) |
+| PSS Memory | The Proportional Set Size (PSS) for a process (The total size of pages it has in main memory (RAM), where each page is divided by the number of processes sharing it). This metric helps to calculate the total memory occupied by multiple processes more accurately. | ![PSS Memory](/images/yp/metrics110.png) |
```
Comment (ddhodge) [inline docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:109]:
```suggestion
| PSS Memory | The proportional set size (PSS) for a process. This is the total size of pages the process has in main memory (RAM), where each page is divided by the number of processes sharing it. This metric helps to calculate the total memory occupied by multiple processes more accurately. | ![PSS Memory](/images/yp/metrics110.png) |
```

File: docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md (hunk 1 above)
Comment (yorq) [reply docs/content/v2024.2/yugabyte-platform/alerts-monitoring/anywhere-metrics.md:101]: These metrics are also available for onprem universes, not only AWS,GCP, Azure. So in general it should be like "non-k8s universes"

**Review by yorq — CHANGES_REQUESTED** [review]
Looks good, the only comment is on availability (should be non-k8s)

**Review by ddhodge — APPROVED** [review]

**Review by yorq — APPROVED** [review]

**Pull request merged**